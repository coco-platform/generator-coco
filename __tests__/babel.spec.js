/**
 * @description - coco:babel generator
 * @author - huang.jian <hjj491229492@hotmail.com>
 */

// packages
const path = require('path');
const fs = require('fs');
const yeomon = require('yeoman-test');
const assert = require('yeoman-assert');
const Renderer = require('../babel/renderer');

describe('coco:babel renderer', () => {
  it('should render proper babel presets within node', () => {
    const answers = {
      presets: ['@babel/preset-env', '@babel/preset-typescript'],
      environment: 'Node',
    };
    const rc = new Renderer(answers);

    rc.renderBabelPresets();

    expect(rc.dependencies).toMatchSnapshot();
    expect(rc.presetsRenderBlock).toMatchSnapshot();
  });

  it('should render proper babel plugins within node', () => {
    const answers = {
      presets: ['@babel/preset-env', '@babel/preset-typescript'],
      environment: 'Node',
    };
    const rc = new Renderer(answers);

    rc.renderBabelPlugins();

    expect(rc.dependencies).toMatchSnapshot();
    expect(rc.pluginsRenderBlock).toMatchSnapshot();
  });

  it('should render proper babel presets within web', () => {
    const answers = {
      presets: ['@babel/preset-env', '@babel/preset-typescript'],
      environment: 'Web',
    };
    const rc = new Renderer(answers);

    rc.renderBabelPresets();

    expect(rc.dependencies).toMatchSnapshot();
    expect(rc.presetsRenderBlock).toMatchSnapshot();
  });

  it('should render proper babel plugins within web', () => {
    const answers = {
      presets: ['@babel/preset-env', '@babel/preset-typescript'],
      environment: 'Web',
    };
    const rc = new Renderer(answers);

    rc.renderBabelPlugins();

    expect(rc.dependencies).toMatchSnapshot();
    expect(rc.pluginsRenderBlock).toMatchSnapshot();
    expect(rc.envRenderBlock).toMatchSnapshot();
  });

  it('should render proper babel plugins within react', () => {
    const answers = {
      presets: ['@babel/preset-env', '@babel/preset-typescript'],
      environment: 'React',
    };
    const rc = new Renderer(answers);

    rc.renderBabelPlugins();

    expect(rc.dependencies).toMatchSnapshot();
    expect(rc.pluginsRenderBlock).toMatchSnapshot();
    expect(rc.envRenderBlock).toMatchSnapshot();
  });
});

describe('coco:babel generator', () => {
  it('should output proper babel configuration', () => {
    const babel = path.resolve(__dirname, '../babel');
    const prompts = {
      presets: ['@babel/preset-env', '@babel/preset-typescript'],
      environment: 'React',
    };

    return yeomon
      .run(babel)
      .withPrompts(prompts)
      .then(() => {
        assert.file(['.babelrc']);

        const configuration = fs.readFileSync('.babelrc', 'utf8');

        expect(configuration).toMatchSnapshot();
      });
  });
});
