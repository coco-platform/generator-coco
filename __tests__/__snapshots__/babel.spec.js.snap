// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`coco:babel generator should output proper babel configuration 1`] = `
"{
  \\"presets\\": [
    [
      \\"@babel/preset-env\\",
      {
        \\"modules\\": false,
        \\"loose\\": true
      }
    ],
    \\"@babel/preset-typescript\\"
  ],
  \\"plugins\\": [
    \\"@babel/plugin-proposal-class-properties\\",
    \\"@babel/plugin-syntax-dynamic-import\\",
    \\"@babel/plugin-transform-regenerator\\",
    [
      \\"@babel/plugin-transform-runtime\\",
      {
        \\"helpers\\": true,
        \\"corejs\\": 2
      }
    ],
    [
      \\"babel-plugin-import\\",
      {
        \\"libraryName\\": \\"antd\\",
        \\"libraryDirectory\\": \\"es\\",
        \\"style\\": \\"css\\"
      }
    ]
  ],
  \\"env\\": {
    \\"test\\": {
      \\"plugins\\": [
        \\"@babel/plugin-transform-modules-commonjs\\"
      ]
    },
    \\"development\\": {
      \\"plugins\\": [
        \\"react-hot-loader/babel\\"
      ]
    }
  }
}"
`;

exports[`coco:babel generator should render proper babel plugins within node 1`] = `Array []`;

exports[`coco:babel generator should render proper babel plugins within node 2`] = `Array []`;

exports[`coco:babel generator should render proper babel plugins within react 1`] = `
Array [
  "@babel/plugin-proposal-class-properties",
  "@babel/plugin-syntax-dynamic-import",
  "@babel/plugin-transform-regenerator",
  "@babel/plugin-transform-runtime",
  "@babel/plugin-transform-modules-commonjs",
  "babel-plugin-import",
  "react-hot-loader",
]
`;

exports[`coco:babel generator should render proper babel plugins within react 2`] = `
Array [
  "@babel/plugin-proposal-class-properties",
  "@babel/plugin-syntax-dynamic-import",
  "@babel/plugin-transform-regenerator",
  Array [
    "@babel/plugin-transform-runtime",
    Object {
      "corejs": 2,
      "helpers": true,
    },
  ],
  Array [
    "babel-plugin-import",
    Object {
      "libraryDirectory": "es",
      "libraryName": "antd",
      "style": "css",
    },
  ],
]
`;

exports[`coco:babel generator should render proper babel plugins within react 3`] = `
Object {
  "development": Object {
    "plugins": Array [
      "react-hot-loader/babel",
    ],
  },
  "test": Object {
    "plugins": Array [
      "@babel/plugin-transform-modules-commonjs",
    ],
  },
}
`;

exports[`coco:babel generator should render proper babel plugins within web 1`] = `
Array [
  "@babel/plugin-proposal-class-properties",
  "@babel/plugin-syntax-dynamic-import",
  "@babel/plugin-transform-regenerator",
  "@babel/plugin-transform-runtime",
  "@babel/plugin-transform-modules-commonjs",
]
`;

exports[`coco:babel generator should render proper babel plugins within web 2`] = `
Array [
  "@babel/plugin-proposal-class-properties",
  "@babel/plugin-syntax-dynamic-import",
  "@babel/plugin-transform-regenerator",
  Array [
    "@babel/plugin-transform-runtime",
    Object {
      "corejs": 2,
      "helpers": true,
    },
  ],
]
`;

exports[`coco:babel generator should render proper babel plugins within web 3`] = `
Object {
  "test": Object {
    "plugins": Array [
      "@babel/plugin-transform-modules-commonjs",
    ],
  },
}
`;

exports[`coco:babel generator should render proper babel presets within node 1`] = `
Array [
  "@babel/preset-env",
  "@babel/preset-typescript",
]
`;

exports[`coco:babel generator should render proper babel presets within node 2`] = `
Array [
  Array [
    "@babel/preset-env",
    Object {
      "targets": Object {
        "node": "10.0.0",
      },
    },
  ],
  "@babel/preset-typescript",
]
`;

exports[`coco:babel generator should render proper babel presets within web 1`] = `
Array [
  "@babel/preset-env",
  "@babel/preset-typescript",
]
`;

exports[`coco:babel generator should render proper babel presets within web 2`] = `
Array [
  Array [
    "@babel/preset-env",
    Object {
      "loose": true,
      "modules": false,
    },
  ],
  "@babel/preset-typescript",
]
`;
